### PR TITLE
Update Rewards payout status messaging

### PR DESCRIPTION
**Pull request by zenparsing**
Resolves https://github.com/brave/brave-browser/issues/18602

## Submitter Checklist:

- [ ] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

See [figma](https://www.figma.com/file/XnadxdmrOHHPYRCdHFD91b/Rewards-Settings?node-id=712%3A50878) for UX designs.

### Scenario 1: Payment pending messaging

- Given that user has ad earnings for the previous month
- And the current day of month is before the payout day
- When the user opens the panel, the NTP, or the rewards page
- Then the user should see the "payment pending" message.

### Scenario 1: Payment processing messaging

- Given that user has ad earnings for the previous month
- And the current day of month is on or after the payout day and less than 4 days after the payout day
- And the user has not received an Ad grant this month
- When the user opens the panel, the NTP, or the rewards page
- Then the user should see the "payment processing" message.

### Scenario 3: Payment arrived messaging

- Given that user has ad earnings for the previous month
- And the current day of month is on or after the payout day and less than 4 days after the payout day
- And the user has received an Ad grant this month
- When the user opens the panel, the NTP, or the rewards page
- Then the user should see the "payment arrived" message.

### CODE REVIEW

File: components/brave_new_tab_ui/components/default/rewards/index.tsx [hunk 1]
```diff
@@ -104,6 +106,7 @@ export const RewardsWidget = createWidget((props: RewardsProps) => {
       nextPaymentDate={adsInfo ? adsInfo.nextPaymentDate : 0}
       earningsThisMonth={adsInfo ? adsInfo.earningsThisMonth : 0}
       earningsLastMonth={adsInfo ? adsInfo.earningsLastMonth : 0}
+      earningsRecieved={earningsReceived}
```
Comment (emerick) [inline components/brave_new_tab_ui/components/default/rewards/index.tsx:109]: Nit: Looks like the property name has a typo (`earningsRecieved`).

File: components/brave_rewards/resources/shared/components/newtab/rewards_card.tsx [hunk 2]
```diff
@@ -85,6 +85,7 @@ interface Props {
   nextPaymentDate: number
   earningsThisMonth: number
   earningsLastMonth: number
+  earningsRecieved: boolean
```
Comment (emerick) [inline components/brave_rewards/resources/shared/components/newtab/rewards_card.tsx:88]: Nit: Same typo as above.

File: components/brave_rewards/resources/shared/components/payment_status_view.tsx [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+  if (days < 1) {
+    return ''
+  }
+
+  return pendingDaysFormatter.format(days)
+}
+
+function getPaymentMonth () {
+  let date = new Date()
+  let lastMonth = new Date(date.getFullYear(), date.getMonth() - 1)
```
Comment (emerick) [inline components/brave_rewards/resources/shared/components/payment_status_view.tsx:74]: Seems like these could be `const` if you want...

File: components/brave_new_tab_ui/components/default/rewards/index.tsx (hunk 1 above)
Comment (zenparsing) [reply components/brave_new_tab_ui/components/default/rewards/index.tsx]: Fixed

File: components/brave_rewards/resources/shared/components/payment_status_view.tsx (hunk 3 above)
Comment (zenparsing) [reply components/brave_rewards/resources/shared/components/payment_status_view.tsx]: Fixed

File: components/brave_rewards/resources/shared/components/newtab/rewards_card.tsx (hunk 2 above)
Comment (zenparsing) [reply components/brave_rewards/resources/shared/components/newtab/rewards_card.tsx]: Fixed

**Review by emerick — APPROVED** [review]
LGTM

**Comment by zenparsing** [comment]
arrived-ntp
arrived-page
arrived-panel-creator
arrived-panel-summary
pending-ntp
pending-page
pending-panel-creator
pending-panel-summary
processing-ntp
processing-page
processing-panel-creator
processing-panel-summary

**Pull request merged**